responsibleai/responsibleai/managers/data_balance_manager.py

```diff
@@ -0,0 +1,321 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the Data Balance Manager class."""
+
+import json
+import warnings
+from pathlib import Path
+from typing import Any, List, Optional
+
+import pandas as pd
+
+from responsibleai._internal.constants import DataBalanceManagerKeys as Keys
+from responsibleai._internal.constants import ListProperties, ManagerNames
+from responsibleai._tools.data_balance.data_balance import DataBalance
+from responsibleai._tools.shared.backends import SupportedBackend, get_spark
+from responsibleai._tools.shared.state_directory_management import \
+    DirectoryManager
+from responsibleai.managers.base_manager import BaseManager
+
+DATA_JSON = "data.json"
+MANAGER_JSON = "manager.json"
+MEASURES_JSON = "measures.json"
+
+
+class DataBalanceManager(BaseManager):
+
+    """Defines the DataBalanceManager for finding
+    data balance measures on a dataset."""
+
+    def __init__(
+        self,
+        target_column: Optional[str],
+        train: Optional[pd.DataFrame],
+        test: Optional[pd.DataFrame],
+    ):
+        """Creates a DataBalanceManager object.
+
+        :param target_column: The name of the target column.
+        :type target_column: str
+        :param train: The training data. This is optional if
+            `custom_data` is provided in `add()`.
+        :type train: pd.DataFrame
+        :param test: The test data. This is optional if
+            `train` is provided in the constructor or
+            `custom_data` is provided in `add()`.
+        :type test: pd.DataFrame
+        """
+        self._target_column = target_column
+        self._train = train
+        self._test = test
+        self._is_added = False
+
+        self._df = None
+        self._backend = None
+        if self._train is not None or self._test is not None:
+            self._df = pd.concat([self._train, self._test])
+            self._backend = self._infer_backend()
+
+        # Populated in add()
+        self._cols_of_interest = None
+        self._pos_label = None
+
+        # Populated in compute()
+        self._data_balance_measures = None
+
+    def add(
+        self,
+        cols_of_interest: List[str],
+        pos_label: Optional[str] = None,
+        custom_data: Optional[Any] = None,
```

Comment:
I'm not sure I follow. We have a dataset via train and test. In which scenario would I want to pass more data?
I may just not have thought of the case you're thinking of which is totally fine, but the API reference should be able to explain it to me which it currently doesn't seem to be.

Reply:
I included a `custom_data` parameter for scenarios where:
- You want to run data balance on your full dataset. I noticed from many sample notebooks that the `train` and `test` data are usually sampled.
- If the dataset is a `spark.sql.DataFrame`. We mainly did this so we can use [the Spark Data Balance](https://microsoft.github.io/SynapseML/docs/features/responsible_ai/Data%20Balance%20Analysis/) to compute measures.

Reply:
I believe the sampling is a choice due to lack of scalability in the dashboard code. @gaugup is looking into resolving this the next few months AFAIK.

It just seems unintuitive from a user perspective to have another data param, especially with a name that doesn't indicate what it's for. Why not pass the spark SQL DF as train and test? @imatiach-msft may have thoughts, too.